poc_mobile_verifier/src/heartbeats.rs

```diff
@@ -110,25 +119,9 @@ impl FromIterator<Heartbeat> for Heartbeats {
     where
         T: IntoIterator<Item = Heartbeat>,
     {
-        let heartbeats = iter
-            .into_iter()
-            .flat_map(|hb| {
-                if hb.validity != proto::HeartbeatValidity::Valid {
-                    return None;
-                }
-                Some((
-                    HeartbeatKey {
-                        hotspot_key: hb.hotspot_key,
-                        cbsd_id: hb.cbsd_id,
-                    },
-                    HeartbeatValue {
-                        reward_weight: hb.reward_weight,
-                        timestamp: hb.timestamp,
-                    },
-                ))
-            })
-            .collect();
-        Self { heartbeats }
+        let mut heartbeats = Self::default();
+        heartbeats.extend(iter);
```

Comment:
clever; are there any appreciable performance implications to this over just `collect` ing into the collection since I have to imagine the _intention_ behind the Extend trait is to add to a collection that already has elements in it?

Reply:
There shouldn't be - this is what we would do in FromIterator anyway